app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -454,7 +470,7 @@ def find_readable_repository
       return nil unless params[:repository]
       repository = Repository.docker_type.find_by(container_repository_name: params[:repository])
       if require_user_authorization?(repository)
-        repository = Repository.readable_docker_catalog.find_by(container_repository_name: params[:repository])
+        repository = Repository.readable_docker_catalog(@host).find_by(container_repository_name: params[:repository])
```

Comment:
I hit this error when trying to pull content:

```
2025-05-12T18:19:06 [E|kat|ce6fbdb0] NoMethodError: undefined method `find_by' for [#<Katello::Repository id: 2, ... ]
```

It looks like readable_docker_catalog is just returning an array and it can't be treated as an active record relation.